bin/cron/push_latest_rds_backup_to_secondary_account

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env ruby
+
+Dir.chdir(File.expand_path('../../../dashboard', __FILE__))
+require_relative '../../dashboard/config/environment'
+require 'aws-sdk'
+require 'cdo/hip_chat'
+
+# This script pushes backup snapshots of our production database into a
+# write-only backup account, with the following steps:
+#
+# 1) Copy the latest automated snapshot to a temporary manual one
+# 2) Wait until ready, and share the manual snapshot with the backup acct
+# 3) On the backup account, copy the shared snapshot to a manual one
+# 4) Wait until ready, and finally go back to the main account and delete the temp snapshot
+#
+# Credential sets must be set up with the names 'default' and 'backup' in the aws config directory,
+# which requires setting appropriate secrets in our Chef config (which will come through via crontab.erb)
+#
+
+def get_aws_account_id(credentials)
+  # based on https://gist.github.com/thomaswitt/3952303, updated for new Aws SDK
+  begin
+    iam = Aws::IAM::Client.new(credentials: credentials)
+    iam = iam.get_user
+    id, user = iam[:user][:arn].match('^arn:aws:iam::([0-9]{12}):(.*)$').captures
+  rescue Aws::IAM::Errors::AccessDenied
+    # This happens because in our case, we have some accounts that are so locked down
+    # that they don't even have IAM get_user permissions. But AWS still spits the
+    # user back in the error message. This is gross, but there doesn't seem to be
+    # any other way.
+    result = $!
+    id, user = result.to_s.match('^User: arn:aws:iam::([0-9]{12}):(.*) is not auth.*$').captures
+  end
+  # rubocop:disable Performance/RedundantMatch
+  raise 'incorrect account id' unless id.match(/^[0-9]{12}$/)
+  # rubocop:enable Performance/RedundantMatch
+
+  [id, user]
+end
+
+def wait_for_snapshot_copy(snapshot)
+  snapshot.wait_until(max_attempts: 120, delay: 5) { |snap|
+    snap.status == 'available'
+  }
+rescue Aws::Waiters::Errors::WaiterFailed
+  HipChat.developers "Timed out waiting for RDS snapshot #{snapshot.snapshot_id} to be available, aborting snapshot backup"
+  exit(1)
+end
+
+class SnapshotNotSharedError < StandardError
+end
+
+def find_shared_snapshot_on_backup(rds_backup, temp_snapshot_name)
+  shared_snapshots = []
+  attempts = 0
+  max_attempts = 60
+  until shared_snapshots.count > 0
+    sleep(1)
+    attempts += 1
+    if attempts > max_attempts
+      raise SnapshotNotSharedError, 'Could not find shared snapshot after 60 seconds'
+    end
+    shared_snapshots = rds_backup.db_snapshots({
+      max_records: 100,
+      include_shared: true,
+    }).select { |snap|
+      snap.snapshot_id.start_with?('arn:aws:rds:') && snap.snapshot_id.end_with?(temp_snapshot_name)
+    }
+  end
+  shared_snapshots.first
+end
+
+def share_snapshot_with_account(credentials_backup, latest_snapshot)
+  # Copy the automated backup into a shareable manual one
+  temp_snapshot_name = 'temp-snapshot-' + Time.now.to_i.to_s
```

Comment:
Can this be simplified to `"temp-snapshot-#{Time.now.to_i.to_s}"`?

IIRC, the `to_s` then becomes redundant.